src/temperature/crud.py

```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Optional
+from fastapi import HTTPException
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from datetime import datetime
+from sqlalchemy.future import select
+import httpx
+from src.temperature.models import Temperature
+from src.city.models import City
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+API_KEY = os.getenv("WEATHER_API_KEY")
+URL = os.getenv("BASE_URL")
+
+
+async def update_temperature(db: AsyncSession):
+    result = await db.execute(select(City))
+    cities = result.scalars().all()
+
+    temperatures = []
+
+    async with httpx.AsyncClient() as client:
+        for city in cities:
+            params = {
+                "key": API_KEY,
+                "q": city.name
+            }
+            response = await client.get(URL, params=params)
```

Comment:
Consider adding error handling for the HTTP request to manage potential exceptions, such as network errors, which could occur during the request.